test/test_auth.py

```diff
@@ -72,6 +72,36 @@ def test_auth_dbname_with_auto_database(bouncer):
     bouncer.test(dbname="postgres", user="someuser", password="anypasswd")
 
 
+def test_unregistered_auto_dbname_with_auto_database(bouncer):
+    with bouncer.ini_path.open() as f:
+        original = f.read()
+    with bouncer.ini_path.open("w") as f:
+        # uncomment the auto-database line
+        new = re.sub(r"^;\* = ", "* = ", original, flags=re.MULTILINE)
+        print(new)
+        f.write(new)
+    # configure the auth_dbname to a database that is not configured
+    # expected behavior is to fallback to auto-database and auto-register.
+    bouncer.admin("set auth_dbname=postgres")
+    bouncer.admin("reload")
+    bouncer.admin("set auth_user='pswcheck'")
+    bouncer.admin(f"set auth_type='md5'")
+    # postgres is not defined in test.ini
```

Comment:
Would be good to ensure that this is the case (e.g by checking the contents of the ini file). This seems like something we might accidentally change in the future.

Reply:
Shall we create a database for this purpose, something like `unconfigured_auth_database` to leave `postgres` somewhat reserved?

Reply:
added new database and improved the validation with [57e51cabc1f24ac1276743093eb086306b94a34e ](https://github.com/pgbouncer/pgbouncer/pull/921/commits/57e51cabc1f24ac1276743093eb086306b94a34e)